### PR TITLE
feat(RHINENG-2218): Hide edge hosts by default when feature flag enabled

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-2218](https://issues.redhat.com/browse/RHINENG-2218).
It makes it so that when the "hbi.api.hide-edge-by-default" feature flag is active, Edge hosts are hidden by default. This means that Edge hosts are excluded from all host requests, except when the request specifically filters on host_type=edge.

I know the Jira says it doesn't have to be behind a feature flag, but part of the Edge Parity plan is to fully integrate them into inventory like normal hosts; so, I think it makes sense to be able to switch it off when that eventually happens.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [x] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: tests/test_xjoin.py [hunk 1]
```diff
@@ -2178,7 +2178,12 @@ def test_spf_owner_id_invalid_field_value(subtests, graphql_query_empty_response
 
 
 # system_profile host_type tests
-def test_query_hosts_filter_spf_host_type(mocker, subtests, graphql_query_empty_response, patch_xjoin_post, api_get):
+# Make sure that regardless of the "hide edge hosts" flag, only the intended SP filters are used
+@pytest.mark.parametrize("hide_edge_ff_value", (True, False))
+def test_query_hosts_filter_spf_host_type(
+    mocker, subtests, graphql_query_empty_response, patch_xjoin_post, api_get, hide_edge_ff_value
```
Comment (thearifismail) [inline tests/test_xjoin.py:2184]: `patch_xjoin_post` is not needed.

File: tests/test_xjoin.py [hunk 2]
```diff
@@ -2211,9 +2216,12 @@ def test_query_hosts_filter_spf_host_type(mocker, subtests, graphql_query_empty_
                 graphql_query_empty_response.reset_mock()
 
 
+# Make sure that regardless of the "hide edge hosts" flag, only the intended SP filters are used
+@pytest.mark.parametrize("hide_edge_ff_value", (True, False))
 def test_query_hosts_filter_spf_host_type_multiple(
-    mocker, subtests, graphql_query_empty_response, patch_xjoin_post, api_get
+    mocker, subtests, graphql_query_empty_response, patch_xjoin_post, api_get, hide_edge_ff_value
```
Comment (thearifismail) [inline tests/test_xjoin.py:2222]: `patch_xjoin_post` not needed.

File: tests/test_xjoin.py [hunk 3]
```diff
@@ -2246,6 +2254,67 @@ def test_query_hosts_filter_spf_host_type_multiple(
             graphql_query_empty_response.reset_mock()
 
 
+# Test feature flag that automatically hides edge hosts
+def test_query_hosts_feature_flag_filter_host_type(
+    mocker, subtests, graphql_query_empty_response, patch_xjoin_post, api_get
```
Comment (thearifismail) [inline tests/test_xjoin.py:2259]: `subtests` and `patch_xjoin_host` not needed.

File: tests/test_xjoin.py [hunk 4]
```diff
@@ -2246,6 +2254,67 @@ def test_query_hosts_filter_spf_host_type_multiple(
...
+            "filter": ({"OR": mocker.ANY}, {"spf_host_type": {"eq": None}}),
+            "fields": mocker.ANY,
+        },
+        mocker.ANY,
+    )
+
+
+# Test that the feature flag hides edge hosts when non-host-type SP filters are provided
+def test_query_hosts_edge_feature_flag_other_sp_filters(
+    mocker, subtests, graphql_query_empty_response, patch_xjoin_post, api_get
```
Comment (thearifismail) [inline tests/test_xjoin.py:2283]: `patch_xjoin_post` not needed

**Review by thearifismail — COMMENTED** [review]
minor comments

**Review by thearifismail — APPROVED** [review]

**Review by fstavela — APPROVED** [review]
LGTM

**Pull request merged**